### PR TITLE
Implement useTaskManager Custom Hook

### PR DESCRIPTION
**Pull request by mannyv123**
## Description

This PR introduces the 'useTaskManager' custom hook, which encapsulates the logic for managing tasks and their interactions with the API. This hook helps in separating concerns, making the code more modular, and improving maintainability.

## Changes Made

- Created the useTaskManager custom hook to manage tasks and their interactions with the API.
- Modified the existing components to use the useTaskManager hook for managing tasks.
- 'useTaskManager.ts' contains the implementation of the 'useTaskManager' custom hook.

## Motivation

- Improved separation of concerns: The logic for managing tasks is now centralized in the 'useTaskManager' hook, making components cleaner and focused on rendering.
- Reusability: The 'useTaskManager' hook can be reused across components to handle tasks consistently.
- Readability: The codebase becomes more organized, making it easier for developers to understand and maintain.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,25 @@
 import './App.css';
-import { ChangeEvent, FormEvent, useEffect, useRef, useState } from 'react';
-import { addTask, deleteSingleTask, getTasks, updateTask } from './utils/api';
-import { Task } from './utils/types';
+import { ChangeEvent, FormEvent, useRef, useState } from 'react';
 import TaskList from './components/TaskList/TaskList';
 import DeleteModal from './components/DeleteModal/DeleteModal';
+import useTaskManager from './hooks/useTaskManager';
```
Comment (tearingItUp786) [inline src/App.tsx]: Styling 💅: don't use a default export for hooks, use a named export.

File: src/hooks/useTaskManager.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  getTasks,
+  updateTask,
+} from '../utils/api';
+
+function useTaskManager() {
+  const [tasks, setTasks] = useState<Task[]>([]);
+
+  useEffect(() => {
+    fetchTasks()
+      .then(() => {})
```
Comment (tearingItUp786) [inline src/hooks/useTaskManager.ts]: Question 🤔: you're not doing anything after you fetch the data? This reads weird.

File: src/hooks/useTaskManager.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      .catch((err) => {
+        console.error('Error getting tasks: ', err);
+      });
+  }, []);
+
+  //Fetch tasks from API
+  const fetchTasks = async () => {
```
Comment (tearingItUp786) [inline src/hooks/useTaskManager.ts]: Question 🤔: why don't we move this outside of this hook and pass it a callback?Will make this easier to test in the future as well.

**Review by tearingItUp786 — CHANGES_REQUESTED** [review]
I want to see some tests set up for this hook. Please create a test for the `useTaskManager.ts` when possible. If you need help, let me know.

File: src/App.tsx (hunk 1 above)
Comment (mannyv123) [reply src/App.tsx]: Updated to named export

File: src/hooks/useTaskManager.ts (hunk 2 above)
Comment (mannyv123) [reply src/hooks/useTaskManager.ts]: Originally had it to wait for fetchTasks to run on mount and thought I needed .then() block. Removed now as a part of following comment response: [Review Comment for fetchTasks()](https://github.com/mannyv123/todo-list/pull/17#discussion_r1310951284)

File: src/hooks/useTaskManager.ts (hunk 3 above)
Comment (mannyv123) [reply src/hooks/useTaskManager.ts]: updated to be outside hook and passed as a callback

**Comment by mannyv123** [comment]
tests added for custom hook and comments addressed

File: src/tests/useTaskManager.test.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { describe, it } from 'vitest';
+import { act, renderHook, waitFor } from '@testing-library/react';
+import { useTaskManager } from '../hooks/useTaskManager';
+import { mockPosts } from './mocks/handlers';
+
```
Comment (tearingItUp786) [inline src/tests/useTaskManager.test.ts]: ✨ Suggestion:  This test should be `colocated` with the `useTaskManager.ts` hook; they should be in the same directory.

File: src/tests/mocks/handlers.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import { rest } from 'msw';
+
+const API_BASE_URL = 'http://localhost:3001';
+
+//Mock data
```
Comment (tearingItUp786) [inline src/tests/mocks/handlers.ts:5]: 🚀 nice:  don't know if this actually mocks your API but good job setting up msw!

File: src/tests/TaskItem.test.tsx [hunk 6]
```diff
@@ -1,43 +1,43 @@
...
+
+// describe('TaskItem', () => {
+//   it('renders task item correctly', () => {
+//     const task = {
+//       _id: '1',
+//       task: 'Example task',
+//       completed: false,
+//       createdAt: new Date(),
+//       updatedAt: new Date(),
+//     };
```
Comment (tearingItUp786) [inline src/tests/TaskItem.test.tsx]: 🤔 Question: why did you comment out this test?

File: src/tests/TaskList.test.tsx [hunk 7]
```diff
@@ -1,69 +1,69 @@
...
-  ];
+// describe('TaskList', () => {
+//   const tasks = [
+//     {
+//       _id: '1',
+//       task: 'Task 1',
+//       completed: false,
+//       createdAt: new Date(),
+//       updatedAt: new Date(),
+//     },
```
Comment (tearingItUp786) [inline src/tests/TaskList.test.tsx]: 🤔 Question: why did you comment out this test?

File: src/tests/mocks/handlers.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  },
+  {
+    _id: '3',
+    task: 'Task 3',
+    completed: true,
+    createdAt: '2023-09-02T20:44:24.326Z',
+    updatedAt: '2023-09-02T20:44:24.326Z',
+  },
+];
+console.log('handlers ran');
```
Comment (tearingItUp786) [inline src/tests/mocks/handlers.ts]: ✨ Suggestion: get rid of this log; you should log at the set up stuff for msw

File: .eslintignore [hunk 9]
```diff
@@ -1,4 +1,5 @@
 package-lock.json
 package.json
 tailwind.config.js
-postcss.config.js
\ No newline at end of file
+postcss.config.js
+/tests/
```
Comment (tearingItUp786) [inline .eslintignore]: why are you ignoring your tests for lining...?

File: src/tests/useTaskManager.test.ts (hunk 4 above)
Comment (mannyv123) [reply src/tests/useTaskManager.test.ts]: moved to same location

File: src/tests/mocks/handlers.ts (hunk 5 above)
Comment (mannyv123) [reply src/tests/mocks/handlers.ts:5]: thank ya thank ya

File: src/tests/mocks/handlers.ts (hunk 8 above)
Comment (mannyv123) [reply src/tests/mocks/handlers.ts]: removed

File: src/tests/TaskItem.test.tsx (hunk 6 above)
Comment (mannyv123) [reply src/tests/TaskItem.test.tsx]: updated the test

File: src/tests/TaskList.test.tsx (hunk 7 above)
Comment (mannyv123) [reply src/tests/TaskList.test.tsx]: updated the test now

File: .eslintignore (hunk 9 above)
Comment (mannyv123) [reply .eslintignore]: changed to include now

**Comment by mannyv123** [comment]
comments addressed and ready for re-review

**Review by tearingItUp786 — APPROVED** [review]
LGTM

**Pull request merged**